src/App.tsx

```diff
@@ -1,9 +1,37 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
+interface Event {
+  target: {
+    value: string;
+  }
+}
+
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [visibleMovies, setVisibleMovies] = useState(moviesFromServer);
+
+  const handleQueryChange = (event: Event) => {
+    const { value } = event.target;
+
+    setQuery(value);
+  };
+
+  useEffect(() => {
+    setVisibleMovies(moviesFromServer.filter(movie => {
+      const searchQuery = query.trim().toLowerCase();
+
+      const movieInfo = `
+        ${movie.title}
+        ${movie.description}
+      `;
+
+      return movieInfo.toLowerCase().includes(searchQuery);
+    }));
+  }, [query]);
```

Comment:
Let's filter on `moviesFromServer` and use variable with result in render.
You don't need `useEffect` for this.
Move out `searchQuery` from filter to avoid re-creating every iteration.
Let's check title and description separately inside filter and return result.

```suggestion
    const searchQuery = query.trim().toLowerCase();
    setVisibleMovies(moviesFromServer.filter(movie => {
      
      const isMovieTitleMatch = movie.title.toLowerCase().includes(searchQuery);
      ...
    
      return isMovieTitleMatch || ...
    }));
```